sdk/ai/ai-inference-rest/src/tracingPolicy.ts

```diff
@@ -0,0 +1,252 @@
+        headers: {
+          _headersMap: {
+          },
+        },
+        method: "POST",
+        timeout: 0,
+        multipartBody: undefined,
+        formData: undefined,
+        disableKeepAlive: false,
+        proxySettings: undefined,
+        streamResponseStatusCodes: undefined,
+        withCredentials: false,
+        abortSignal: undefined,
+        tracingOptions: {
+        },
+        onUploadProgress: undefined,
+        onDownloadProgress: undefined,
+        requestId: "e89d3930-257d-4d96-a60b-216e5ecc8040",
+        allowInsecureConnection: false,
+        enableBrowserStreams: true,
+      }
+
+       */
+      if (!request.body)
+        return next(request);
+      const body = JSON.parse(request.body as string);
+
+      const spanAttributes: TracingAttributes = {
+        [TracingAttributesEnum.Operation_Name]: "operation name", ///TODD: don't know how to determine yet
+        [TracingAttributesEnum.Request_Model]: body.model,
+        [TracingAttributesEnum.Request_Frequency_Penalty]: body.frequency_penalty,
+        [TracingAttributesEnum.Request_Max_Tokens]: body.max_tokens,
+        [TracingAttributesEnum.Request_Presence_Penalty]: body.presence_penalty,
+        [TracingAttributesEnum.Request_Stop_Sequences]: body.stop,
+        [TracingAttributesEnum.Request_Temperature]: body.temperature,
+        [TracingAttributesEnum.Request_Top_K]: body.top_k,
+        [TracingAttributesEnum.Request_Top_P]: body.top_p
+      };
+
+      const { span, tracingContext } = tryCreateSpan(tracingClient, request, spanAttributes) ?? {};
+
+      if (!span || !tracingContext) {
+        return next(request);
+      }
+
+      try {
+        const response = await tracingClient.withContext(tracingContext, next, request);
+        tryProcessResponse(span, response);
+        return response;
+      } catch (err: any) {
+        tryProcessError(span, err);
+        throw err;
+      }
+    },
+  };
+}
+
+function tryCreateTracingClient(): TracingClient | undefined {
+  try {
+    return createTracingClient({
+      namespace: "",
+      packageName: "@azure-rest/ai-inference",
+      packageVersion: SDK_VERSION,
+    });
+  } catch (e: unknown) {
+    logger.warning(`Error when creating the TracingClient: ${getErrorMessage(e)}`);
+    return undefined;
+  }
+}
+
+function tryCreateSpan(
+  tracingClient: TracingClient,
+  request: PipelineRequest,
+  spanAttributes: Record<string, unknown>,
+): { span: TracingSpan; tracingContext: TracingContext } | undefined {
+  try {
+    // As per spec, we do not need to differentiate between HTTP and HTTPS in span name.
+    const { span, updatedOptions } = tracingClient.startSpan(
+      "{gen_ai.operation.name} {gen_ai.request.model}", ///TODO: not sure how to retrieve the name yet
+      { tracingOptions: {} },  ///TODO: should I pass tracing options?
```

Comment:
I think you can use the tracingOptions from the request `request.tracingOptions`

Reply:
Done